File: src/backend/logProcessor.js
const fs = require('fs');
const path = require('path');

function processLogs() {
  const logPath = path.join(__dirname, '../logs/sasl1.log');
  try {
    const data = fs.readFileSync(logPath, 'utf8');
    const lines = data.split('\n');

    const logs = lines.map(line => {
      if (line.trim() === '') return null;

      const regex = /^(\w+\s+\d+\s+\d+:\d+:\d+)\s+\w+\s+\w+\/\w+\/\w+\[(\d+)\]:\s+warning:\s+([^\[]+)\[(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\]:\s+(.*)$/;
      const match = line.match(regex);

      if (match) {
        const [_, timestamp, processId, host, ip, message] = match;

        return {
          timestamp, 
          processId, 
          host: host.trim(), 
          ip,
          message: message.trim() 
        };
      } else {
        console.error('Log parsing failed for line:', line);
        return null;
      }
    }).filter(entry => entry !== null); 

    return logs;
  } catch (error) {
    console.error('Error reading log file:', error);
    return [];
  }
}

module.exports = { processLogs };

File: src/backend/log_processor.py
import ipapi
import re

def get_geolocation(ip):
    try:
        data = ipapi.location(ip=ip)
        return data.get("country_name", "Unknown")
    except Exception as e:
        print(f"Error fetching geolocation for IP {ip}: {e}")
        return "Unknown"
    
def process_logs(log_file_path):
    logs = []
    with open(log_file_path, 'r') as file:
        lines = file.readlines()
        for line in lines:
            if line.strip() == '':
                continue

            match = re.match(
                r"^(\w+\s+\d+\s+\d+:\d+:\d+)\s+\w+\s+\w+\/\w+\/\w+\[(\d+)\]:\s+warning:\s+([^\[]+)\[(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\]:\s+(.*)$", 
                line
            )

            if match:
                timestamp, process_id, host, ip, message = match.groups()
                
                # geolocation = get_geolocation(ip)  

                logs.append({
                    "timestamp": timestamp,
                    "processId": process_id,
                    "host": host.strip(),
                    "ip": ip,
                    # "geolocation": geolocation,
                    "message": message.strip()
                })
            else:
                print("Log parsing failed for line:", line)

    return logs
